frappe/desk/doctype/form_tour/form_tour.js

```diff
@@ -125,3 +162,75 @@ async function get_first_document(doctype) {
 
 	return docname || "new";
 }
+
+async function get_path(frm) {
+	let route = [frm.doc.view_name];
+	switch (route[0]) {
+		case "Workspaces":
+			frm.doc.list_name = "";
+			frm.doc.new_document_form = 0;
+			frm.doc.report_name = "";
+			frm.doc.page_name = "";
+			frm.doc.dashboard_name = "";
+			frm.doc.reference_doctype = "";
```

Comment:
clearing fields you can handle this in the view_name method. just clear all fields if the view_name value is changed

Reply:
some cases like  `frm.doc.report_name = ""` depends on list_name as well. i can add this to respective field's method but this is less likely to fail as it will clear everything before saving. if you think that would still be better let me know!